src/provider/states/creating_config.rs

```diff
@@ -0,0 +1,503 @@
+        // TODO: refactor this to use the map created above
+        let referenced_config_maps = CreatingConfig::get_config_maps(pod).await;
+
+        // Check if all required config maps have been created in the api-server
+        // Transition pod to retry state if some are missing or we geta kube error when
+        // communicating with the api server
+        let config_map_data = match CreatingConfig::retrieve_config_maps(
+            client.clone(),
+            pod.namespace(),
+            referenced_config_maps,
+        )
+        .await
+        {
+            Ok(config_maps) => config_maps,
+            Err(MissingConfigMapsError {
+                missing_config_maps,
+            }) => {
+                warn!(
+                    "Unable to find all required config maps for service {}, missing: {:?}",
+                    pod_state.service_name, &missing_config_maps
+                );
+                return Transition::next(
+                    self,
+                    WaitingConfigMap {
+                        missing_config_maps,
+                    },
+                );
+            }
+            Err(e) => {
+                // Not sure, shouldn't really happen, just do what we know: wait
+                return Transition::next(
+                    self,
+                    WaitingConfigMap {
+                        missing_config_maps: vec![format!(
+                            "An unexepected error occurred: {:?}",
+                            e
+                        )],
+                    },
+                );
+            }
+        };
+
+        // At this point we have all config maps and their content that we need, otherwise the
+        // error cases in the above match statement would have moved the pod to the waiting for
+        // configmap state already
+
+        let template_data = if let Ok(data) = CreatingConfig::create_render_data(&pod_state) {
+            data
+        } else {
+            error!("Unable to parse directories for command template as UTF8");
+            return Transition::next(
+                self,
+                SetupFailed {
+                    message: "Unable to parse directories for command template as UTF8".to_string(),
+                },
+            );
+        };
+
+        // Write the config files
+        let config_dir = pod_state.get_service_config_directory();
+        for (target_path, volume) in volume_mounts {
+            let volume = volume.clone();
+            let joined_target_path = config_dir.join(&target_path);
+            debug!("Applying config map {} to {}", volume, target_path);
+            if let Some(volume_content) = config_map_data.get(&volume) {
+                if let Err(e) = CreatingConfig::apply_config_map(
+                    volume_content,
+                    joined_target_path,
+                    &template_data,
+                ) {
+                    // Creation of config file failed!
```

Comment:
"e" is unused and shouldn't you do something here?

(I have a feeling I commented here before but not sure)

Reply:
Your feeling is correct and I have the corresponding feeling that I already fixed this ...